### ISSUE TITLE
How does validation work?

### ISSUE THREAD

**Issue by sensationTI** [AlexeyAB/darknet#6432]
Because of all images in the validation set need to be labeled just like the ones in the training set, I assume yolo validates the result by predicting bounding box on the validation set, and then compare these bounding boxes with the labeled ones. Correct me if I am wrong. Thanks.

Is there any way that I can generate a testing result using some kind of testing set?

**Comment by stephanecharette** [comment]
I assume the same way as you that is how validation works.

Normally, you take X% of your images for training, and Y% of your images for validation.  Usually, the images are randomly chosen by whatever tool or bash script, there isn't a specific way to choose them.  For most of my projects, I split all my labelled images into 80% and 20%.  There are a few projects where I have a very large number of labelled images, and in those cases I adjust the % down to limit the validation set to around 1000 images.

For generating the different sets, I use DarkMark which has a simple slider to pick a percentage:  https://www.ccoderun.ca/darkmark/Summary.html#DarkMarkDarknet

**Comment by stephanecharette** [comment]
I did some tests a while back with the following:

- 50% training, 50% validation
- 70% training, 30% validation
- 85% training, 15% validation
- 99% training, 1% validation

If you have a non-trivial number of labelled images, there is very little difference between all these.  The big difference is actually making sure that max_batches is high enough.  See the results here:  https://www.ccoderun.ca/programming/2020-01-04_neural_network_training/#Details

**Comment by sensationTI** [comment]
> I assume the same way as you that is how validation works.
> 
> Normally, you take X% of your images for training, and Y% of your images for validation. Usually, the images are randomly chosen by whatever tool or bash script, there isn't a specific way to choose them. For most of my projects, I split all my labelled images into 80% and 20%. There are a few projects where I have a very large number of labelled images, and in those cases I adjust the % down to limit the validation set to around 1000 images.
> 
> For generating the different sets, I use DarkMark which has a simple slider to pick a percentage: https://www.ccoderun.ca/darkmark/Summary.html#DarkMarkDarknet

Thanks, but I am wondering if I can generate a mAP result for my testing set just like the mAP for validation set when you use the command for example `darknet.exe detector map data/obj.data yolo-obj.cfg backup\yolo-obj_7000.weights`

**Comment by stephanecharette** [comment]
> Thanks, but I am wondering if I can generate a mAP result for my testing set just like the mAP for validation set when you use the command for example `darknet.exe detector map data/obj.data yolo-obj.cfg backup\yolo-obj_7000.weights`

Not that I know of.

**Comment by sensationTI** [comment]
> > Thanks, but I am wondering if I can generate a mAP result for my testing set just like the mAP for validation set when you use the command for example `darknet.exe detector map data/obj.data yolo-obj.cfg backup\yolo-obj_7000.weights`
> 
> Not that I know of.

Thank you so much!
